src/main/scala/net/snowflake/spark/snowflake/Utils.scala

```diff
@@ -71,8 +71,11 @@ object Utils {
   /**
     * Client info related variables
     */
-  private[snowflake] lazy val sparkAppName =
-    SparkContext.getOrCreate().getConf.get("spark.app.name", "")
+  private[snowflake] lazy val sparkAppName = if (SparkEnv.get != null) {
+    SparkEnv.get.conf.get("spark.app.name", "")
```

Comment:
Is this enough to uniquely identify a Spark run? Do we expect to use this combined with timestamps?

Reply:
This is only a general information field. It is not unique.
We'll use "spark.app.id" as the unique id lately.